src/main/java/dev/latvian/mods/kubejs/util/MobEffectUtil.java

```diff
@@ -0,0 +1,46 @@
+package dev.latvian.mods.kubejs.util;
+
+import dev.latvian.mods.kubejs.typings.Info;
+import net.minecraft.core.Holder;
+import net.minecraft.world.effect.MobEffect;
+import net.minecraft.world.effect.MobEffectInstance;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+
+public class MobEffectUtil {
+	@Contract("_ -> new")
+	@Info("Creates a new instance from the old one")
+	public static @NotNull MobEffectInstance of(MobEffectInstance oldInstance) {
+		return new MobEffectInstance(oldInstance)
+	}
+
+    @Contract("_ -> new")
+	@Info("Creates an instance for the given effect")
+    public static @NotNull MobEffectInstance of(Holder<MobEffect> effect) {
+        return new MobEffectInstance(effect);
+    }
+
+    @Contract("_, _ -> new")
+	@Info("Creates an instance for the given effect and duration")
+    public static @NotNull MobEffectInstance of(Holder<MobEffect> effect, int duration) {
+        return new MobEffectInstance(effect, duration);
+    }
+
+    @Contract("_, _, _ -> new")
+	@Info("Creates an instance for the given effect, duration and amplifier")
+    public static @NotNull MobEffectInstance of(Holder<MobEffect> effect, int duration, int amplifier) {
+        return new MobEffectInstance(effect, duration, amplifier);
+    }
+
+    @Contract("_, _, _, _, _ -> new")
+	@Info("Creates an instance for the given effect, duration, amplifier, ambient, and visible to the HUD")
+    public static @NotNull MobEffectInstance of(Holder<MobEffect> effect, int duration, int amplifier, boolean ambient, boolean visible) {
+        return new MobEffectInstance(effect, duration, amplifier, ambient, visible);
+    }
+
+    @Contract("_, _, _, _, _, _ -> new")
+	@Info("Creates an instance for the given effect, duration, amplifier, ambient, visible to the HUD, and to show the icon")
```

Comment:
Show the icon where? Inventory? HUD? Second Monitor? or

Reply:
I do not know what the icon is. It's a thing in Minecraft and I have zero idea what it dose